src/snowflake/cli/plugins/nativeapp/project_model.py

```diff
@@ -185,3 +186,13 @@ def debug_mode(self) -> Optional[bool]:
         if self.definition.application:
             return self.definition.application.debug
         return None
+
+    def get_bundle_context(self) -> BundleContext:
+        return BundleContext(
+            self.package_name,
+            self.artifacts,
+            self.project_root,
+            self.bundle_root,
+            self.deploy_root,
+            self.generated_root,
+        )
```

Comment:
is it possible to pass these by name? If we get the order wrong we might not notice given how the test is written. Before this call and the definition were side by side, but now they're in different files so IMHO ordering is more of a concern.

Reply:
BTW, this is not really what I meant by "extract a bundlecontext from NAPM". What I had in mind was that BundleContext would become the authority, and NAPM would own a (immutable) bundlecontext to which it would delegate the appropriate property lookups.

Reply:
Changed to named params.

Yeah I started by doing something similar, but I actually think it's much simpler for NAPM to hold the logic of how to construct a BundleContext, instead of BundleContext being aware of anywhere its being used. WDYT?

Reply:
sure, I'm fine either way